src/components/auth/profile/DonationTable.tsx

```diff
@@ -0,0 +1,184 @@
+import {
+  Card,
+  Checkbox,
+  darken,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TextField,
+  Grid,
+  Table,
+  TableRow,
+  Avatar,
+  Button,
+} from '@mui/material'
+import StarIcon from '@mui/icons-material/Star'
+import { DatePicker, LocalizationProvider } from '@mui/x-date-pickers'
+import { money } from 'common/util/money'
+import { format, isAfter, isBefore, parseISO } from 'date-fns'
+import React, { useEffect, useState } from 'react'
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward'
+import { useTranslation } from 'next-i18next'
+import { UserDonation } from 'gql/donations'
+import theme from 'common/theme'
+import { bg, enUS } from 'date-fns/locale'
+import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns'
+import styled from '@emotion/styled'
+import { Box } from '@mui/system'
+
+export type DonationTableProps = {
+  donations: UserDonation[] | undefined
+}
+const CheckboxLabel = styled.label``
+function DonationTable({ donations }: DonationTableProps) {
+  const { t, i18n } = useTranslation()
+  const [fromDate, setFromDate] = React.useState<Date | null>(null)
+  const [toDate, setToDate] = React.useState<Date | null>(null)
+  const [monthly, setMonthly] = React.useState(true)
+  const [oneTime, setOneTime] = React.useState(true)
+  const [filteredDonations, setFilteredDonations] = useState(donations)
+  useEffect(() => {
+    setFilteredDonations(donations)
+  }, [donations])
+  useEffect(() => {
+    if (monthly && oneTime) {
+      setFilteredDonations(donations)
+      return
+    }
+    if (!monthly && !oneTime) {
+      setFilteredDonations([])
+    }
+    if (monthly) {
+      setFilteredDonations(donations?.filter((d) => d.type !== 'donation'))
+    }
+    if (oneTime) {
+      setFilteredDonations(donations?.filter((d) => d.type === 'donation'))
+    }
+  }, [monthly, oneTime])
```

Comment:
It looks like you need `useMemo` instead of `useState` + `useEffect` here

```tsx
const filteredDonations = useMemo(() => {
   // other conditions here
  if (monthly) {
      return donations?.filter((d) => d.type !== 'donation')
    }
    if (oneTime) {
      return donations?.filter((d) => d.type === 'donation')
    }
    return donations
}, [donations, monthly, oneTime])
```

Reply:
Check out https://github.com/podkrepi-bg/frontend/pull/719/commits/84188cfe9667c25e2371834303743601fd88649d